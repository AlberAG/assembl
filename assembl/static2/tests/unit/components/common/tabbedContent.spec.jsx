import React from 'react';
import renderer from 'react-test-renderer';

import TabbedContent from '../../../../js/app/components/common/tabbedContent';

describe('TabbedContent component', () => {
  const tabs = [
    {
      id: 'tab1',
      title: 'One',
      text: 'If we override the transmitter, we can get to the PNG bus through the solid state RSS alarm!'
    },
    { id: 'tab2', title: 'Two', text: 'I\'ll input the auxiliary RSS monitor, that should feed the ADP monitor!' },
    {
      id: 'tab3',
      title: 'Three',
      text: 'The ADP matrix is down, quantify the neural protocol so we can program the COM firewall!'
    }
  ];
  const renderBody = (tab, idx) => (
    <div className="box">
      <h2>
        {tab.title} ({idx + 1})
      </h2>
      <p>{tab.text}</p>
    </div>
  );
  it('should match snapshot', () => {
    const rendered = renderer.create(<TabbedContent tabs={tabs} renderBody={renderBody} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('should match snapshot with divClassName', () => {
    const rendered = renderer.create(<TabbedContent divClassName="right" tabs={tabs} renderBody={renderBody} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('should match snapshot with bodyRowClassName', () => {
    const rendered = renderer.create(<TabbedContent bodyRowClassName="right" tabs={tabs} renderBody={renderBody} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});