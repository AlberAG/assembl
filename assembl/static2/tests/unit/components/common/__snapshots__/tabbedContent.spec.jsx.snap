// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabbedContent component should match snapshot 1`] = `
<div
  className="tabbed-content"
>
  <div
    className="row"
  >
    <div
      className="col-md-4 col-xs-12"
    >
      <a
        className="tab-title-active ellipsis"
        onClick={[Function]}
      >
        One
      </a>
    </div>
    <div
      className="col-md-4 col-xs-12"
    >
      <a
        className="tab-title ellipsis"
        onClick={[Function]}
      >
        Two
      </a>
    </div>
    <div
      className="col-md-4 col-xs-12"
    >
      <a
        className="tab-title ellipsis"
        onClick={[Function]}
      >
        Three
      </a>
    </div>
  </div>
  <div
    className="tab-body row"
  >
    <div
      className="box"
    >
      <h2>
        One
         (
        1
        )
      </h2>
      <p>
        If we override the transmitter, we can get to the PNG bus through the solid state RSS alarm!
      </p>
    </div>
  </div>
</div>
`;

exports[`TabbedContent component should match snapshot with bodyRowClassName 1`] = `
<div
  className="tabbed-content"
>
  <div
    className="row"
  >
    <div
      className="col-md-4 col-xs-12"
    >
      <a
        className="tab-title-active ellipsis"
        onClick={[Function]}
      >
        One
      </a>
    </div>
    <div
      className="col-md-4 col-xs-12"
    >
      <a
        className="tab-title ellipsis"
        onClick={[Function]}
      >
        Two
      </a>
    </div>
    <div
      className="col-md-4 col-xs-12"
    >
      <a
        className="tab-title ellipsis"
        onClick={[Function]}
      >
        Three
      </a>
    </div>
  </div>
  <div
    className="tab-body right row"
  >
    <div
      className="box"
    >
      <h2>
        One
         (
        1
        )
      </h2>
      <p>
        If we override the transmitter, we can get to the PNG bus through the solid state RSS alarm!
      </p>
    </div>
  </div>
</div>
`;

exports[`TabbedContent component should match snapshot with divClassName 1`] = `
<div
  className="tabbed-content right"
>
  <div
    className="row"
  >
    <div
      className="col-md-4 col-xs-12"
    >
      <a
        className="tab-title-active ellipsis"
        onClick={[Function]}
      >
        One
      </a>
    </div>
    <div
      className="col-md-4 col-xs-12"
    >
      <a
        className="tab-title ellipsis"
        onClick={[Function]}
      >
        Two
      </a>
    </div>
    <div
      className="col-md-4 col-xs-12"
    >
      <a
        className="tab-title ellipsis"
        onClick={[Function]}
      >
        Three
      </a>
    </div>
  </div>
  <div
    className="tab-body row"
  >
    <div
      className="box"
    >
      <h2>
        One
         (
        1
        )
      </h2>
      <p>
        If we override the transmitter, we can get to the PNG bus through the solid state RSS alarm!
      </p>
    </div>
  </div>
</div>
`;
